build_tools/bazel_to_cmake/bazel_to_cmake_targets.py

```diff
@@ -233,37 +233,33 @@ def convert_target(target):
   if target.startswith("//llvm-external-projects/iree-dialects"):
     return _convert_iree_dialects_target(target)
 
+  if target.startswith("@"):
+    raise KeyError(f"No conversion found for target '{target}'")
+
   # IREE root paths map to package names based on explicit rules.
   # If changing these, make the corresponding change in iree_macros.cmake
   # (iree_package_ns function).
 
   # Map //compiler/src/iree/(.*) -> iree::\1 (i.e. iree::compiler::\1)
   m = re.match("^//compiler/src/iree/(.+)", target)
   if m:
-    return ["iree::" + _convert_bazel_path(m.group(1))]
+    return ["iree::" + _convert_to_cmake_path(m.group(1))]
 
   # Map //runtime/src/iree/(.*) -> iree::\1
   m = re.match("^//runtime/src/iree/(.+)", target)
   if m:
-    return ["iree::" + _convert_bazel_path(m.group(1))]
-
-  # Map //runtime/bindings/(.*) -> iree::bindings::\1
-  m = re.match("^//runtime/bindings/(.+)", target)
-  if m:
-    return ["iree::bindings::" + _convert_bazel_path(m.group(1))]
-
-  # Map //samples/(.*) -> iree::samples::\1
-  m = re.match("^//samples[/|:](.+)", target)
-  if m:
-    return ["iree::samples::" + _convert_bazel_path(m.group(1))]
+    return ["iree::" + _convert_to_cmake_path(m.group(1))]
 
   # Map //tools/(.*) -> \1
   m = re.match("^//tools[/|:](.+)", target)
   if m:
-    return [_convert_bazel_path(m.group(1))]
+    return [_convert_to_cmake_path(m.group(1))]
```

Comment:
I still don't really understand why we're the tools be in the global namespace like this when we already have aliases for them that do the same thing.

Reply:
Aliases are weird. They don't show up in the list of targets and can't be referenced directly in some CMake functions. It's nicer to just pick the primary name we want the targets to have and use that instead.

Reply:
That's fair. We have cc_library targets in there as well though. Previously we were only aliasing binaries

Reply:
Idea: maybe we could have a tag that marks a target as "global", which elevates it to the global namespace? Then we don't have all cc_binary targets being global (which especially gets weird with benchmark tests) and we don't have tool libraries as global

Reply:
> That's fair. We have cc_library targets in there as well though. Previously we were only aliasing binaries

Only one library remains in that directory: `iree_opt_main`... which is getting named `_iree_opt_main` now... errr...

Reply:
Okay, the `iree_opt_main` library is actually no longer needed, since https://github.com/google/iree/commit/9228670a0e06cf52589f141f7fb171fe574089d7 and https://github.com/google/iree/commit/fe19812aa5187f9951e6fa42127a60ce07b423a7 (`custom-opt` and `iree-tf-opt` used to link that library in, but they provide their own `*_opt_main.cc` files now). After folding that library into its only use (`iree-opt`), we now just have binaries in `tools/`.

Reply:
Maybe add a note in the BUILD file that only binaries should go in tools? I think we should also drop our global aliases in cc_binary, since this was basically the reason they existed and we've got other binaries that really shouldn't be in the global namespace (like random benchmarks)

Reply:
SG. I expanded on the comments and dropped the global binary aliases.